packages/react/src/components/AccountSettings/ChangePassword/ChangePassword.tsx

```diff
@@ -0,0 +1,107 @@
+import React from 'react';
+
+import { Logger } from 'aws-amplify';
+import { changePassword, translate } from '@aws-amplify/ui';
+
+import { useAuth } from '../../../internal';
+import { View, Flex } from '../../../primitives';
+import {
+  DefaultCurrentPassword,
+  DefaultError,
+  DefaultNewPassword,
+  DefaultSubmitButton,
+} from './defaultComponents';
+import { ChangePasswordProps } from './types';
+import { FormValues } from '../types';
+
+const logger = new Logger('ChangePassword');
+
+const ChangePassword: React.ComponentType<ChangePasswordProps> = (props) => {
+  const { onSuccess, onError } = props;
+  const [error, setError] = React.useState<string>(null);
+  const [formValues, setFormValues] = React.useState<FormValues>({});
+
+  const { user, isLoading: isAuthUserLoading } = useAuth();
+
+  /** Translations */
+  // TODO: add AccountSettingsTextUtil to collect these strings
+  const currentPasswordLabel = translate('Current Password');
+  const newPasswordLabel = translate('New Password');
+  const updatePasswordText = translate('Update password');
+
+  /** Auth API Handler */
+  const handleChangePassword = React.useCallback(async () => {
+    const { currentPassword, newPassword } = formValues;
+    try {
+      await changePassword({ user, currentPassword, newPassword });
+      setError(null);
+
+      if (onSuccess) {
+        onSuccess(); // notify success to the parent
+      }
```

Comment:
Could just use optional chaining here (`onSuccess?.()`), same with the error handler below

Reply:
Done in https://github.com/aws-amplify/amplify-ui/commit/97936eec8e4fce7511c6c3c5deee9aed98b9984b:

https://github.com/aws-amplify/amplify-ui/blob/97936eec8e4fce7511c6c3c5deee9aed98b9984b/packages/react/src/components/AccountSettings/ChangePassword/ChangePassword.tsx#L61-L66